enum.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+# Microverse-202-Advanced-Building-Blocks-Enumerables
+# By Jair Jaramillo
+
+# module enum
+module Enumerable
+  def bubble_sort(array)
+    (array.size - 1).times do |x|
+      array[x], array[x + 1] = array[x + 1], array[x] if array[x] > array[x + 1]
+    end
+    (array.size - 1).times do |x|
+      bubble_sort(array) if array[x] > array[x + 1]
+    end
+    array
+  end
+
+  def bubble_sort_by(arra)
+    (arra.size - 1).times do |x|
+      arra[x], arra[x + 1] = arra[x + 1], arra[x] if yield(arra[x], arra[x + 1]).positive?
+    end
+    (arra.size - 1).times do |x|
+      next unless (arra[x].length - arra[x + 1].length).positive?
+
+      bubble_sort_by(arra) do |_left, _right|
+        yield(arra[x], arra[x + 1])
+      end
+    end
+    arra
+  end
+
+  def my_each
+    return to_enum unless block_given?
+
+    counter = 0
+    while counter < to_a.length
+      yield to_a[counter]
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    counter = 0
+    while counter < to_a.length
+      yield to_a[counter], counter
+      counter += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    i = 0
+    while i < to_a.length
+      result << to_a[i] if yield to_a[i]
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = false)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif pattern.class == Class
+      my_each { |x| return false unless x.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |x| return false unless x =~ pattern }
+    elsif pattern.nil?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless x == pattern }
+    end
+    true
+  end
+
+  def my_any?(pattern = false)
+    if block_given?
+      my_each { |x| return true if yield(x) }
+    elsif pattern.class == Class
+      my_each { |x| return true if x.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |x| return true if x =~ pattern }
+    elsif pattern.nil?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if x == pattern }
+    end
+    false
+  end
```

Comment:
* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil